### PR TITLE
Initial update draft

### PR DESCRIPTION
**Pull request by sfc-gh-mheavin**
### Pre-review checklist
   * [ ] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [ ] I've added new automated tests to verify correctness of my new code.
   * [ ] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the section below.

### Changes description

I updated the Readme to reflect the changes for creating connections. I also added a note that links to the PrPr registration form, per Anurag's request.
...

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,68 +78,101 @@ Use SnowCLI to build a function or stored procedure, or create a streamlit if yo
...
 
-```ini
-[connections.test]
-account = $SNOWCLI_ACCOUNT
-user = $SNOWCLI_USER
-password = $SNOWCLI_PASSWORD
-```
 
-will be replaced with environment variables:
+#### Use environment variables for Snowflake credentials
```
Comment (sfc-gh-astus) [inline README.md:148]: We decided to use solution from python connector and remove our. Their replacing works a bit differently, we have to provide list of possible environment variable names. It should be done in few days, so please hold off on that part for a while

File: README.md [hunk 2]
```diff
@@ -75,68 +78,101 @@ Use SnowCLI to build a function or stored procedure, or create a streamlit if yo
...
+
+1. If you already specify your credentials in a configuration file, as shown, you can remove them from the
+   connection settings.
+
+   ```
+   [connections.test]
+   account = $SNOWCLI_ACCOUNT
+   user = $SNOWCLI_USER
+   password = $SNOWCLI_PASSWORD
+   ```
```
Comment (sfc-gh-turbaszek) [inline README.md:174]: Today it work a bit different. If users have the following config:
```
key0 = value0

[section]
[section.option]
key1 = value1

[section2]
key2 = value2
```

They can override existing configuration by variables like:
```
SNOWFLAKE_KEY0
SNOWFLAKE_SECTION_OPTION_KEY1
SNOWFLAKE_SECTION2_KEY2
```

Comment (sfc-gh-mheavin) [reply README.md]: Updated

**Review by sfc-gh-anugupta — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -75,68 +80,113 @@ Use SnowCLI to build a function or stored procedure, or create a streamlit if yo
...
+   For example, to add a Snowflake connection called **myconnection** with the credentials: account **myaccount**,
+   user profile **johndoe**, and password **hunter2**,
+   add the following lines to the configuration file:
 
-1. Create a `.snowsql` folder with a `config` file at the following path: `%USERPROFILE%\.snowsql\config`
-2. Add a new configuration for your Snowflake connection with SnowCLI. You must prefix the configuration with `connections.`.
+   ```
+   [connections]
+   [connections.myconnection]
+   accountname = myaccount
```
Comment (sfc-gh-pjob) [inline README.md:134]:
```suggestion
   account = myaccount
```

`accountname` was changed to `account` - the same for other examples of connection configuration

File: README.md [hunk 4]
```diff
@@ -75,68 +80,113 @@ Use SnowCLI to build a function or stored procedure, or create a streamlit if yo
...
+   username = jondoe
```
Comment (sfc-gh-pjob) [inline README.md:135]:
```suggestion
   user = jondoe
```
`username` was changed `user` - the same for other examples of connection configuration

File: README.md [hunk 5]
```diff
@@ -75,68 +80,113 @@ Use SnowCLI to build a function or stored procedure, or create a streamlit if yo
...
+
+
+To add credentials as environment variables:
+
+1. Define the following environment variables, as appropriate for your operating system:
+
+   ```
+   [connections]
+   [connections.myconnection]
+   accountname = SNOWFLAKE_CONNECTIONS_MYCONNECTION_ACCOUNT
```
Comment (sfc-gh-pjob) [inline README.md:182]: Currently environment variables are not used in the config file at all.

You can set environment variables (in your system/shell, outside of the config file) and they will override values set in configuration. Or environment variables can be combined with the values from the config file.

Examples:

1)) Combination of environment variables and the config file:
Config file:
```
[connections]
[connections.myconnection]
account = myaccount
user = jondoe
```

Environment variable set in system/shell:
```
SNOWFLAKE_CONNECTIONS_MYCONNECTION_PASSWORD=pass1234
```

2)) Overriding values from the config file by environment variables:
```
[connections]
[connections.myconnection]
account = myaccount
user = jondoe
password=xyz000
```

Environment variable set in system/shell:
```
SNOWFLAKE_CONNECTIONS_MYCONNECTION_PASSWORD=pass1234
```

---

Both examples have the same result - password "pass1234".

File: README.md (hunk 3 above)
Comment (sfc-gh-mheavin) [reply README.md:134]: done

File: README.md (hunk 4 above)
Comment (sfc-gh-mheavin) [reply README.md:135]: done

File: README.md (hunk 5 above)
Comment (sfc-gh-mheavin) [reply README.md:182]: Updated

File: README.md [hunk 6]
```diff
@@ -69,78 +73,144 @@ snow --version
...
 
-#### Add credentials manually using Windows
+If you prefer, you can specify Snowflake credentials in system environment variables, instead of specifying them
+in configuration files. You can use environment variables only to replace connection parameters. Environment variables for
+the configuration must use the following formats:
 
-1. Create a `.snowsql` folder with a `config` file at the following path: `%USERPROFILE%\.snowsql\config`
-2. Add a new configuration for your Snowflake connection with SnowCLI. You must prefix the configuration with `connections.`.
+- SNOWFLAKE_\<section-name\>_\<key\>=\<value\>
+- SNOWFLAKE_\<section-name\>_\<option-name\>__\<key\>=\<value\>
```
Comment (sfc-gh-turbaszek) [inline README.md:163]:
```suggestion
- SNOWFLAKE_\<section-name\>_\<option-name\>_\<key\>=\<value\>
```

Comment (sfc-gh-mheavin) [reply README.md:165]: done

Comment (sfc-gh-pjob) [reply README.md:165]: this one is not fixed :)

**Review by sfc-gh-turbaszek — APPROVED** [review]

**Pull request merged**